2do_fe/src/redux/reducers/index.js

```diff
@@ -0,0 +1,41 @@
+import * as types from "../constants/task.constants";
+
+const initialState = {
+  // all tasks
+  tasks: [],
+  loading: true,
+
+  singleTask: null,
+};
+
+const taskReducer = (state = initialState, action) => {
```

Comment:
It makes a lot more sense after the research as well as your explanation of the work flow.